nutils/evaluable.py

```diff
@@ -297,7 +297,7 @@ def _node(self, cache, subgraph, times):
 
   @property
   def _node_details(self):
-    return ''
+    return self.dtype.__name__ if hasattr(self, 'dtype') else ''
```

Comment:
Rather than putting this in the base class and testing for an attribute I'd say proper way is to place this in `Array._node_details`, or better yet `Array._node` which is where the shape is handled as well. This way operations such as `Transpose` and `Einsum` can add details without losing the type.